### PR TITLE
chore(ui): migrate Panel, PanelBody and PanelFooter components to Typescript

### PR DESCRIPTION
**Pull request by guoda-puidokaite**
# Summary

- Migrated `Panel`, `PanelBody` and `PanelFooter` components to `TypeScript`, enhancing code quality, maintainability, and scalability.
- Utilised static typing to minimize runtime errors and improve the development experience.
- Refactored the components for cleaner, more maintainable code.
- Exposed deprecated .js components as `Portal` is used (which is in progress).
- Improved Storybook types and documentation. **NOTE:** Please make sure to check the old code for correct types. Live Storybook is incorrect.
- **NOTE**: Removed `ContentArea` and `ContentAreaWrapper` from Stories already as they're deprecated.

# Changes Made

**TypeScript Migration**:

- Converted `Panel`, `PanelBody` and `PanelFooter` components files to `TypeScript`.
- Added comprehensive type annotations.
- Added old `.js` components to the `deprecated_js` folder.
- Exported deprecated components as `Portal` is used (which is in progress).

**Refactoring**:

- Made necessary adjustments to address TypeScript-detected issues.
- Further refactored for cleaner and more understandable code.

**Testing**:

- Added edge cases:
  - `Panel`
      - Rendering panel without props
      - Rendering panel with undefined className
      - Rendering panel without a heading
      - Clicking close button when panel is already closed
      - Double-click on close button
  - `PanelBody`
      - Render without any props
      - Render with null children
      - Render with undefined children
      - Applies default styles
      - Renders multiple children
      - Renders no children
  - `PanelFooter`
      - Render without any props
      - Applies default styles
      - Renders multiple children
      - Renders no children
- Organised tests by what they check for better readability and maintenance.
- Verified components behaviour using `Storybook`.
- Updated tests to use `vitest`.

**Documentation:**

- Improved `Storybook` documentation types and comments.

**Release:**

- Generated a `changeset` file.

# Related Issues

- Closes https://github.com/cloudoperators/juno/issues/243

# Testing Instructions

1. Check `Storybook`

# Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have made corresponding changes to the documentation (if applicable).
- [x] My changes generate no new warnings or errors.

### CODE REVIEW

File: packages/ui-components/src/index.js [hunk 1]
```diff
@@ -66,9 +66,9 @@ export { ModalFooter } from "./deprecated_js/ModalFooter/index.js"
 export { NativeSelect } from "./components/NativeSelect/NativeSelect.component"
 export { NativeSelectOption } from "./components/NativeSelectOption/NativeSelectOption.component"
 export { NativeSelectOptionGroup } from "./components/NativeSelectOptionGroup/NativeSelectOptionGroup.component"
-export { Panel } from "./components/Panel/index.js"
-export { PanelBody } from "./components/PanelBody/index.js"
-export { PanelFooter } from "./components/PanelFooter/index.js"
+export { Panel } from "./components/Panel/Panel.component"
```
Comment (barsukov) [inline packages/ui-components/src/index.js:69]: **(Breaking change)**
Panel use Portal, Portal is still in migration, I would suggest to export this component stil from deprecated until we will not finish all of Portal depended components.

File: packages/ui-components/src/components/Panel/Panel.stories.tsx [hunk 2]
```diff
@@ -4,11 +4,12 @@
  */
 
 import React from "react"
-import { Panel } from "./index.js"
-import { PanelBody } from "../PanelBody/index.js"
-import { ContentAreaWrapper } from "../ContentAreaWrapper/index.js"
-import { ContentArea } from "../ContentArea/index.js"
-import { PortalProvider } from "../../deprecated_js/PortalProvider/PortalProvider.component.js"
+
```
Comment (barsukov) [inline packages/ui-components/src/components/Panel/Panel.stories.tsx:7]: We shoudl fix this types all story :) make it typed

**Review by barsukov — CHANGES_REQUESTED** [review]
Great job and few breaking changes :)

File: packages/ui-components/src/components/Panel/Panel.stories.tsx (hunk 2 above)
Comment (guoda-puidokaite) [reply packages/ui-components/src/components/Panel/Panel.stories.tsx:7]: Still a draft, working on it locally.

**Comment by guoda-puidokaite** [comment]
Hey, @barsukov. I've marked this PR as a draft. 
Having issues with the stories, need to find a solution once I've some time after training. 
The blocking change is very helpful though 🙏. Might even be related to the issue I'm facing. Thank You for the early review!

File: packages/ui-components/src/index.js (hunk 1 above)
Comment (guoda-puidokaite) [reply packages/ui-components/src/index.js:69]: Very helpful, thanks!

**Review by barsukov — APPROVED** [review]
LGTM!

**Review by ArtieReus — CHANGES_REQUESTED** [review]
reviewing :)

File: packages/ui-components/src/components/Panel/Panel.stories.tsx [hunk 3]
```diff
@@ -17,24 +16,32 @@ export default {
...
+} as Meta
 
-const Template = (args) => (
-  <ContentAreaWrapper>
+const Template: StoryFn<PanelProps> = (args) => (
+  <div>
     <Panel {...args}>
       <PanelBody>Panel Body Content</PanelBody>
     </Panel>
-    <ContentArea className="dummy-css-ignore jn-h-[150px]">Content Area</ContentArea>
```
[the remark this answers is not in the document]
Comment (guoda-puidokaite) [reply packages/ui-components/src/components/Panel/Panel.stories.tsx:36]: If you open the component, you can see it's deprecated. So decided to remove these deprecated components from Story already. Good question though, updated PR description. 🙈

File: packages/ui-components/src/components/Panel/Panel.component.tsx [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+  jn-px-8
+`
+
+const panelTitleStyles = `
+  jn-text-theme-high
+  jn-text-lg
+  jn-font-bold
+`
+
+type PanelSize = "default" | "large"
```
[the remark this answers is not in the document]
Comment (guoda-puidokaite) [reply packages/ui-components/src/components/Panel/Panel.component.tsx:60]: https://github.com/cloudoperators/juno/pull/606#discussion_r1860452312

File: packages/ui-components/src/components/Panel/Panel.stories.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (ArtieReus) [reply packages/ui-components/src/components/Panel/Panel.stories.tsx:36]: Ahh ok! Meaning also that there is no replacement for this or?

**Review by ArtieReus — APPROVED** [review]
lgtm

**Review by barsukov — APPROVED** [review]
LGTM

**Pull request merged**